cumulus/polkadot-omni-node/lib/src/common/spec.rs

```diff
@@ -92,6 +98,31 @@ fn warn_if_slow_hardware(hwbench: &sc_sysinfo::HwBench) {
 	}
 }
 
+pub(crate) trait InitBlockImport<Block: BlockT, RuntimeApi> {
+	type BlockImport: sc_consensus::BlockImport<Block> + Clone + Send + Sync;
+	type ExtraReturnValue;
+
+	fn init_block_import(
+		client: Arc<ParachainClient<Block, RuntimeApi>>,
+	) -> sc_service::error::Result<(Self::BlockImport, Self::ExtraReturnValue)>;
+}
+
+pub(crate) struct ClientBlockImport;
+
+impl<Block: BlockT, RuntimeApi> InitBlockImport<Block, RuntimeApi> for ClientBlockImport
+where
+	RuntimeApi: Send + ConstructNodeRuntimeApi<Block, ParachainClient<Block, RuntimeApi>>,
+{
+	type BlockImport = Arc<ParachainClient<Block, RuntimeApi>>;
+	type ExtraReturnValue = ();
```

Comment:
nit: maybe `BlockImportAuxiliaryData` would be a better naming?
And also generic type in `ParachainService` could be `BIAuxiliaryData`?